### PR TITLE
Feature/ornl ci conda

### PR DESCRIPTION
**Pull request by RamonAra209**
- Created a ORNL CI configuration for Conda on Ascent
- Modified SDK_WORK_DIR to reflect ORNL [standard](https://github.com/ExaWorks/SDK/tree/master/.gitlab#environment-variables)
- Forced usage of python 3.8.10

### CODE REVIEW

File: .gitlab/ornl-ci-conda.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    PYTHON_MODULE: "python/3.8-anaconda3"
+    CONDA_ARCH: "ppc64le" 
+
+# .on_summit:
+#   variables:
+#     SITE: "summit"
+#     VENV_ENV_NAME: "summit-env"
+#     CONDA_WORK_DIR: ${SDK_WORK_DIR_ALPINE}/conda
+#     PYTHON_MODULE: "python/3.8-anaconda3"
+#     CONDA_ARCH: "x86_64"
```
Comment (mtitov) [inline .gitlab/ornl-ci-conda.yml:27]: `CONDA_ARCH: "ppc64le"` (just to be consistent, even it is not used now)

File: .gitlab/ornl-ci-conda.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    RUNNER_TYPE: "batch"
+    SCHEDULER_PARAMETERS: "-P CSC449 -nnodes 1 -W 45"
+  needs: [ conda_build_ascent ]
+  extends: [ .on_ascent, .job_tags, .conda_test ]
+
+
+# ALWAYS
+conda_cleanup_ascent:
+  stage: cleanup
+  when: always
```
Comment (mtitov) [inline .gitlab/ornl-ci-conda.yml:126]: these lines could be removed since they are in `.conda_cleanup`

File: .gitlab/ornl-ci-conda.yml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  variables:
+    RUNNER_TYPE: "nobatch"
+  extends: [ .on_ascent, .job_tags .conda_cleanup ]
```
Comment (mtitov) [inline .gitlab/ornl-ci-conda.yml:129]: missed comma: `extends: [ .on_ascent, .job_tags, .conda_cleanup ]`

File: .gitlab/ornl-ci-conda.yml (hunk 1 above)
Comment (RamonAra209) [reply .gitlab/ornl-ci-conda.yml]: Done

File: .gitlab/ornl-ci-conda.yml (hunk 2 above)
Comment (RamonAra209) [reply .gitlab/ornl-ci-conda.yml]: Done

File: .gitlab/ornl-ci-conda.yml (hunk 3 above)
Comment (RamonAra209) [reply .gitlab/ornl-ci-conda.yml]: Done, also rebased onto master (since smartsim was merged)

File: .gitlab/ornl-ci-conda.yml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  - build
+  - test
+  - cleanup
+
+
+### MACHINES
+.on_ascent:
+  variables:
+    SITE: "ascent"
+    VENV_ENV_NAME: "ascent-env"
```
Comment (mtitov) [inline .gitlab/ornl-ci-conda.yml:16]: typo, should be `CONDA_ENV_NAME`

File: .gitlab/ornl-ci-conda.yml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    CONDA_WORK_DIR: ${SDK_WORK_DIR_OPEN}/conda
+    PYTHON_MODULE: "python/3.8-anaconda3"
+    CONDA_ARCH: "ppc64le" 
+
+# .on_summit:
+#   variables:
+#     SITE: "summit"
+#     VENV_ENV_NAME: "summit-env"
```
Comment (mtitov) [inline .gitlab/ornl-ci-conda.yml:24]: same here -> `CONDA_ENV_NAME`

File: .gitlab/ornl-ci-conda.yml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  variables:
+    MAINTAINER: "arambula2@llnl.gov"
+    TESTS_GROUP: "conda"
+    SITE_ID: "ornl-$SITE"
+  script:
+    - source ${CONDA_WORK_DIR}/miniconda3-${SITE}/etc/profile.d/conda.sh
+    - eval "$(conda shell.posix hook)"
+    - conda activate ${CONDA_WORK_DIR}/${CONDA_ENV_NAME}
+    - python3 ci/tests/test.py -s
+    - for TEST in flux parsl rp swift-t parsl-flux rp-flux; do
```
Comment (mtitov) [inline .gitlab/ornl-ci-conda.yml:79]: let's also have tests in sync with requirements
`for TEST in parsl rp; do`

p.s. I guess we can keep `flux` if only it is provided with `modules`

Comment (RamonAra209) [reply .gitlab/ornl-ci-conda.yml:79]: Doesn't look like `flux` module is available on lassen, so I'm assuming the same for ascent? I'll push the change. Let me know if you want me to add an explicit `module load <flux>`, and re-enable the test if `flux` is available on Ascent

**Review by mtitov — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**